spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkSQLProperties.java

```diff
@@ -63,4 +63,8 @@ private SparkSQLProperties() {}
   // Controls the WAP branch used for write-audit-publish workflow.
   // When set, new snapshots will be committed to this branch.
   public static final String WAP_BRANCH = "spark.wap.branch";
+
+  public static final String PLANNING_MODE = "spark.sql.iceberg.planning-mode";
```

Comment:
It's a little strange to have a "planning mode" that doesn't affect deletes. I'd say this should be the default for both (with the ability to override it specifically for deletes like we do for other settings) or we should have one for data and one for deletes.

Reply:
We do have two, maybe I should call it `data-planning-mode`?

Reply:
I renamed it.